testing/citest/tests/kube_v2_smoke_test.py

```diff
@@ -0,0 +1,267 @@
+        self.__image_repository,
+        self.__desired_image_tag)
+
+  def create_app(self):
+    """Creates OperationContract that creates a new Spinnaker Application."""
+    contract = jc.Contract()
+    return st.OperationContract(
+        self.agent.make_create_app_operation(
+            bindings=self.bindings, application=self.TEST_APP,
+            account_name=self.bindings['SPINNAKER_KUBERNETES_V2_ACCOUNT']),
+        contract=contract)
+
+  def delete_app(self):
+    """Creates OperationContract that deletes a new Spinnaker Application."""
+    contract = jc.Contract()
+    return st.OperationContract(
+        self.agent.make_delete_app_operation(
+            application=self.TEST_APP,
+            account_name=self.bindings['SPINNAKER_KUBERNETES_V2_ACCOUNT']),
+        contract=contract)
+
+  def deploy_manifest(self):
+    """Creates OperationContract for deployManifest.
+
+    To verify the operation, we just check that the deployment was created.
+    """
+    bindings = self.bindings
+    name = self.TEST_APP + '-deployment'
+    payload = self.agent.make_json_payload_from_kwargs(
+        job=[{
+            'cloudProvider': 'kubernetes',
+            'moniker': {
+                'app': self.TEST_APP
+            },
+            'account': bindings['SPINNAKER_KUBERNETES_V2_ACCOUNT'],
+            'source': 'text',
+            'type': 'deployManifest',
+            'user': '[anonymous]',
+            'manifests': [{
+                'apiVersion': 'apps/v1beta2',
+                'kind': 'Deployment',
+                'metadata': {
+                    'name': name,
+                    'namespace': self.TEST_NAMESPACE,
+                    'labels': {
+                        'app': self.TEST_APP,
+                    }
+                },
+                'spec': {
+                    'replicas': 1,
+                    'selector': {
+                        'matchLabels': {
+                            'app': self.TEST_APP,
+                        }
+                    },
+                    'template': {
+                        'metadata': {
+                            'labels': {
+                                'app': self.TEST_APP,
+                            }
+                        },
+                        'spec': {
+                            'containers': [{
+                                'name': 'primary',
+                                'image': self.__desired_image_id,
+                            }]
+                        }
+                    }
+                }
+            }]
+        }],
+        description='Deploy manifest',
+        application=self.TEST_APP)
+
+    builder = kube.KubeContractBuilder(self.kube_v2_observer)
+    (builder.new_clause_builder('Deployment created',
+                                retryable_for_secs=15)
+     .get_resources(
+         'deploy',
+         extra_args=[name, '--namespace', self.TEST_NAMESPACE])
+     .contains_path_eq('spec/replicas', 1))
```

Comment:
ok, but you could EXPECT a match if you want to look at more stuff.
See EXPECT(ov_factory.value_list_contains(jp.DICT_MATCHES in other tests.

Reply:
Here's an example from google_smoke_test
https://github.com/spinnaker/spinnaker/blob/master/testing/citest/tests/google_smoke_test.py#L193

There it is taking the configuration it passed in, makes a tweak to it to add the name it expects it to create, then passes that as [a subset of] the expected observation in order to verify that the configuration was plumbed through spinnaker and into the actual platform resources created.

I think this is particularly important because mocked unit tests arent going to tell you that the actual API isnt behaving the way you thought and your resources are not configured the way you expected.

This test is sufficient to verify that it got created which is probably normally ok if you can take the config for granted. But since nothing else is testing the config, this would be a good place for it.